src/components/Rate/Rate.jsx

```diff
@@ -0,0 +1,167 @@
+import "./Rate.css";
+
+const signInURL = {
+    pathname: "/sign/in",
+    state: {
+        redirectTo: window.location.pathname
+    }
+};
+
+class Rate extends Component {
+    state = {
+        score: 0,
+        tempScore: 0,
+
+        isLoading: true,
+        me: this.props.me || {},
+        error: false,
+        goToLogin: false,
+        max: 0,
+        numRatings: 0,
+        average: 0,
+        ratings: {
+            5: 0,
+            4: 0,
+            3: 0,
+            2: 0,
+            1: 0
+        }
+    };
+
+    componentDidMount() {
+        const { planId } = this.props;
+
+        return api
+            .get("/rate/" + planId)
+            .then(d => {
+                let ratings = { 1: 0, 2: 0, 3: 0, 4: 0, 5: 0 };
+                const max = d.data.reduce(
+                    (acc, r) => (r.num > acc ? r.num : acc),
+                    0
+                );
+                d.data.map(rating => {
+                    ratings[rating.score] = rating.num;
+                });
+                const numRatings = d.data.length;
+                const average =
+                    d.data.reduce((acc, r) => acc + r.score, 0) / numRatings;
+
+                this.setState({
+                    isLoading: false,
+                    me: d.data.me,
+                    max,
+                    numRatings,
+                    average,
+                    ratings
+                });
+            })
+            .catch(error => this.setState({ error }));
+    }
+
+    setRate = score => {
+        const { planId, me } = this.props;
+        if (!me) {
+            return this.setState({ goToLogin: true });
+        }
+        return api
+            .post("/rate/", { score, plan_id: planId })
+            .then(ratings => {
+                this.setState({ score });
+                this.componentDidMount();
+            })
+            .catch(error => this.setState({ error }));
+    };
+
+    hoverRate = rate => {
+        this.setState({ tempRate: rate });
+    };
+
+    render() {
+        const {
+            me,
+            score,
+            tempRate,
+            goToLogin,
+            max,
+            numRatings,
+            average,
+            ratings
+        } = this.state;
+        const displayRate = tempRate || score;
+        if (goToLogin) {
+            return <Redirect to={signInURL} />;
+        }
+
+        return (
+            <Fragment>
+                עד כמה אתם חושבים שהתוכנית מטיבה עם סביבתה?
+                <div className="row">
+                    <div className="col-5">
+                        <Rating
+                            direction="rtl"
+                            onClick={this.setRate}
+                            onHover={this.hoverRate}
+                            initialRating={score}
+                        />
+                    </div>
+                    <div className="col-7">
+                        {displayRate === 1 &&
```

Comment:
i'd have a const object somewhere at the top and here do something like ratingsMap[displayRate]